File: 백준/Gold/2470. 두 용액/두 용액.swift
let n = Int(readLine()!)!
var arr = readLine()!.split(separator: " ").compactMap {Int(String($0))}
arr = arr.sorted()

var (left, right) = (0, n-1)
var sums = arr[left] + arr[n-1]
var ab = abs(sums)
var ans = [arr[left], arr[right]]

while left < right - 1 {
    if sums > 0 {
        right -= 1
    } else {
        left += 1
    }
    
    sums = arr[left] + arr[right]
  
    if abs(sums) < ab {
        ans[0] = arr[left]
        ans[1] = arr[right]
        ab = abs(sums)
    }
}

print(ans[0], ans[1])

File: 백준/Gold/2470. 두 용액/두 용액.py
import sys
input = sys.stdin.readline

n = int(input())
liquids = list(map(int, input().split()))

left, right = 0, n-1
liquids.sort()
answer = [left, right]
m_val = 10e9

while left < right:
    s_val = liquids[left] + liquids[right]
    
    if abs(s_val) < m_val:
        answer = [left, right]
        m_val = abs(s_val)

    if s_val < 0:  # 음수
        left += 1
    else:
        right -= 1

print(liquids[answer[0]], liquids[answer[1]])